### PR TITLE
Security Update: pythonPackages.python-gnupg: 0.4.3 -> 0.4.4

### PR DESCRIPTION
**Pull request by stigtsp**
###### Motivation for this change

Fixes CVE-2019-6690

https://blog.hackeriet.no/cve-2019-6690-python-gnupg-vulnerability/

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/python-gnupg/default.nix [hunk 1]
```diff
@@ -3,11 +3,11 @@
 buildPythonPackage rec {
   name    = "${pname}-${version}";
```
Comment (dotlambda) [inline pkgs/development/python-modules/python-gnupg/default.nix]: Remove this line.

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/python-gnupg/default.nix [hunk 2]
```diff
@@ -3,11 +3,11 @@
...
-  version = "0.4.3";
+  version = "0.4.4";
 
   src = fetchPypi {
     inherit pname version;
-    sha256 = "2d158dfc6b54927752b945ebe57e6a0c45da27747fa3b9ae66eccc0d2147ac0d";
+    sha256 = "45daf020b370bda13a1429c859fcdff0b766c0576844211446f9266cae97fb0e";
   };
 
   propagatedBuildInputs = [ gnupg1 ];
```
Comment (dotlambda) [inline pkgs/development/python-modules/python-gnupg/default.nix]: It seems this is not needed. Should be removed.

**Review by dotlambda — CHANGES_REQUESTED** [review]
Please mention the fixed vuln (+ link) in the commit description.

**Comment by stigtsp** [comment]
> Please mention the fixed vuln (+ link) in the commit description.

Sure. Would you like a single squashed commit?

**Comment by dotlambda** [comment]
Actually, I think the update and the improvements should be two separate commits so that we can backport only one of them.

**Comment by stigtsp** [comment]
> Actually, I think the update and the improvements should be two separate commits so that we can backport only one of them.

Done

**Comment by dotlambda** [comment]
Thanks a lot!

**Comment by dotlambda** [comment]
Backported in 119a93190a86794179e13ec38d51140e953a8ca9.

**Pull request merged**